src/scenarios/shared/runner.py

```diff
@@ -35,34 +36,107 @@ def parseargs(self):
         '''
         Parses input args to the script
         '''
-        parser = ArgumentParser()
-        subparsers = parser.add_subparsers(title='subcommands for scenario tests', dest='testtype')
-        startupparser = subparsers.add_parser(const.STARTUP)
+        parser = ArgumentParser(description='test.py runs the test with specified commands. Usage: test.py <command> <optional subcommands> <options>',
+                                formatter_class=RawTextHelpFormatter)
+        subparsers = parser.add_subparsers(title='subcommands for scenario tests', 
+                                           dest='testtype')
+
+        # startup command
+        startupparser = subparsers.add_parser(const.STARTUP,
+                                              description='measure time to main of running the project')
         self.add_common_arguments(startupparser)
 
-        sdkparser = subparsers.add_parser(const.SDK)
-        sdkparser.add_argument('sdktype', choices=[const.CLEAN_BUILD, const.BUILD_NO_CHANGE, const.NEW_CONSOLE], type=str.lower)
+        # sdk command
+        sdkparser = subparsers.add_parser(const.SDK, 
+                                          description='subcommands for sdk scenario',
+                                          formatter_class=RawTextHelpFormatter)
+        sdkparser.add_argument('sdktype', 
+                                choices=[const.CLEAN_BUILD, const.BUILD_NO_CHANGE, const.NEW_CONSOLE], 
+                                type=str.lower,
+                                nargs=1,
+                                help= 
+'''
+clean_build:     measure duration of building from source in each iteration
+build_no_change: measure duration of building with existing output in each iteration
+new_console:     measure duration of creating a new console template
+'''
+                               )
         self.add_common_arguments(sdkparser)
 
-        crossgenparser = subparsers.add_parser(const.CROSSGEN)
-        crossgenparser.add_argument('--test-name', dest='testname', type=str, required=True)
-        crossgenparser.add_argument('--core-root', dest='coreroot', type=str, required=True)
+        crossgenparser = subparsers.add_parser(const.CROSSGEN,
+                                               description='measure duration of the crossgen compilation',
+                                               formatter_class=RawTextHelpFormatter)
+        crossgenparser.add_argument('--assembly-name', 
```

Comment:
Why are we changing the argument name here? I am always a little leery of making breaking changes like this.

Reply:
Changing the name to `--assembly-name` because 1) the option actually just represents an assembly name, not the test name 2) it's more consistent with the crossgen2 tests. Thanks for the reminder that it will need to be updated in our test pipeline proj files as well.

Reply:
Actually I'm reconsidering the name change because it requires PR submit in runtime and synchronizition between 2 repos... @billwert  May I ask your opinion on this? Does the name change make sense to you or you'd prefer keeping the original?